mapper/orm/src/main/java/org/hibernate/search/mapper/orm/mapping/impl/HibernateOrmMappingImpl.java

```diff
@@ -49,7 +54,12 @@ public boolean isWorkable(Object entity) {
 
 	private HibernateOrmSearchManagerBuilder createSearchManagerBuilder(EntityManager entityManager) {
 		SessionImplementor sessionImplementor = entityManager.unwrap( SessionImplementor.class );
-		// TODO check that the session refers to the same session factory used when building the mapping
+		SessionFactoryImplementor givenSessionFactory = sessionImplementor.getSessionFactory();
+
+		if ( !givenSessionFactory.getUuid().equals( sessionFactoryImplementor.getUuid() ) ) {
```

Comment:
I think you really should use `.equals` instead. If two different instances of `SessionFactoryImplementor` have the same UUID, something is terribly wrong, and I'd rather have this code fail...

Reply:
No problem to change that. Just for curiosity, is it really possible? :)